### DISCUSSION TITLE
keepUnusedDataFor vs refetchOnMountOrArgChange

### DISCUSSION THREAD

**Discussion by viniciusbsneto** [reduxjs/redux-toolkit#1484] [Q&A] [answered]
I've already read RTK Query documentation but there's something I still don't understand quite well. Considering equal cache validation times for both props, what is the practical difference between `keepUnusedDataFor` and `refetchOnMountOrArgChange` (time in seconds)? Aren't they both going to refetch data after validation time?

**Comment by phryneas** [comment, marked as answer]
`keepUnusedDataFor`: Imagine you unmount your last component using `useMyQuery(5)`. After `keepUnusedDataFor` seconds, the cache value will now be deleted from cache.

`refetchOnMountOrArgChange`: Imagine you already have a component using `useMyQuery(5)`. Now a new component mounts and also uses `useMyQuery(5)`. Normally, that data would not refetch. If `refetchOnMountOrArgChange` is set and more time than that value has passed when that new component mounts, it will re-fetch even though a result is already in the cache.

**Reply by viniciusbsneto** [reply]
Thank you! I think I have it clearer now.
